Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs

```diff
@@ -373,9 +385,50 @@ private void ConfigureUIForLobbyMode(LobbyMode mode)
         /// <summary>
         /// Called directly by UI elements!
         /// </summary>
+        /// <remarks>
+        /// For a reactive character select screen, the local player's selection is presented instantaneously
+        /// before a NetworkListEvent can be received from the server.
+        /// To accomplish this, the previous seat selected will be cleared or re-populated by a shared player, and
+        /// the new seat will be populated with the local selection.
+        /// </remarks>
         /// <param name="seatIdx"></param>
         public void OnPlayerClickedSeat(int seatIdx)
         {
+            if (m_LastSeatSelected != -1)
+            {
+                var otherPlayerSharesSeat = false;
+
+                // if any other user shares seat, set them as owner; else deactivate seat
+                foreach (var lobbyPlayer in CharSelectData.LobbyPlayers)
+                {
+                    if (lobbyPlayer.ClientId == NetworkManager.Singleton.LocalClientId)
+                    {
+                        // ignore self;
+                        continue;
+                    }
+
+                    if (lobbyPlayer.SeatIdx == m_LastSeatSelected)
+                    {
+                        // populate this seat with a shared player's lobby player state
+                        otherPlayerSharesSeat = true;
+                        m_PlayerSeats[m_LastSeatSelected].SetState(lobbyPlayer.SeatState, lobbyPlayer.PlayerNum, lobbyPlayer.PlayerName);
+                    }
+                }
+
+                if (!otherPlayerSharesSeat)
+                {
+                    // no other player shared this seat; it is safe to just disable
+                    m_PlayerSeats[m_LastSeatSelected].SetState(CharSelectData.SeatState.Inactive, -1, string.Empty);
+                }
+            }
+
+            // get local player, and populate the seat that is anticipated to be taken
+            TryGetLobbyPlayer(NetworkManager.Singleton.LocalClientId, out var localLobbyPlayerState);
```

Comment:
might be worth to note this is the client drive part where we take the seat without waiting for the server driven value